### PR TITLE
approach to inherit from abstract superclass

### PR DESCRIPTION
**Pull request by njr-11**
The pull demonstrates the approach to dealing with multiple implementation of incompatible versions of spec API by isolating differences (which are minimal) to an abstract superclass.  The enabled feature selects which superclass to bring in, either the no-op one for Concurrency 3.0 and earlier, or the one for Concurrency 3.1+ which includes a minimal amount of new methods that require a newer Java SE level to implement.

### CODE REVIEW

**Comment by njr-11** [comment]
#build

File: dev/io.openliberty.concurrent.internal.compat31/src/io/openliberty/concurrent/internal/compat/ContextServiceBase.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+package io.openliberty.concurrent.internal.compat;
+
+import java.util.concurrent.Flow.Subscriber;
+
+import com.ibm.wsspi.threadcontext.ThreadContextDescriptor;
+
+import io.openliberty.concurrent.internal.compat.impl.ContextualSubscriber;
+
+/**
+ * Superclass for ContextServiceImpl when using Concurrency 3.0 or less.
```
Comment (mswatosh) [inline dev/io.openliberty.concurrent.internal.compat31/src/io/openliberty/concurrent/internal/compat/ContextServiceBase.java]: This should be Concurrency 3.1 or later, but can be changed in a later pull

**Review by mswatosh — DISMISSED** [review]

File: dev/io.openliberty.concurrent.internal.compat31/src/io/openliberty/concurrent/internal/compat/ContextServiceBase.java (hunk 1 above)
Comment (njr-11) [reply dev/io.openliberty.concurrent.internal.compat31/src/io/openliberty/concurrent/internal/compat/ContextServiceBase.java]: Good catch -- too much copying and pasting there!

**Review by mswatosh — APPROVED** [review]

**Comment by njr-11** [comment]
#build
#libby

**Pull request merged**